lib/src/rules/unreachable_from_main.dart

```diff
@@ -63,11 +63,59 @@ class UnreachableFromMain extends LintRule {
     NodeLintRegistry registry,
     LinterContext context,
   ) {
-    var visitor = _Visitor(this);
+    var visitor = _Visitor(this, context);
     registry.addCompilationUnit(this, visitor);
   }
 }
 
+/// A visitor which gathers all of the top-level and static declarations which
+/// we may wish to report on.
+class _DeclarationGatherer extends SimpleAstVisitor<void> {
```

Comment:
I probably wouldn't add the overhead of a new visitor when it's just iterating over the declarations in a compilation unit, but I can live with it if you don't agree.